Packet++/src/InfiniBandLayer.cpp

```diff
@@ -0,0 +1,209 @@
+#include "InfiniBandLayer.h"
+#include "PayloadLayer.h"
+#include "PacketUtils.h"
+#include "Logger.h"
+#include <string.h>
+#include "EndianPortable.h"
+
+namespace pcpp
+{
+
+    InfiniBandLayer::InfiniBandLayer( uint8_t opcode, int se,
+                    int mig, int pad, uint16_t pkey, uint32_t qpn, int ack_req,
+                    uint32_t psn)
+    {
+        const size_t headerLen = sizeof(rxe_bth);
+		m_DataLen = headerLen;
+		m_Data = new uint8_t[headerLen];
+		memset(m_Data, 0, headerLen);
+		rxe_bth* bthHdr = (rxe_bth*)m_Data;
+
+        bthHdr->opcode = opcode;
+        bthHdr->flags = (pad << 4) & BTH_PAD_MASK;
+        if (se)
+            bthHdr->flags |= BTH_SE_MASK;
+        if (mig)
+            bthHdr->flags |= BTH_MIG_MASK;
+        bthHdr->pkey = htobe16(pkey);
+        bthHdr->qpn = htobe32(qpn & BTH_QPN_MASK);
+        psn &= BTH_PSN_MASK;
+        if (ack_req)
+            psn |= BTH_ACK_MASK;
+        bthHdr->apsn = htobe32(psn);
+        m_Protocol = IB;
+    }
+
+    void InfiniBandLayer::parseNextLayer()
+	{
+        if (m_DataLen <= sizeof(rxe_bth))
+			return;
+
+		uint8_t* bthData = m_Data + sizeof(rxe_bth);
+		size_t bthDataLen = m_DataLen - sizeof(rxe_bth);
+
+        m_NextLayer = new PayloadLayer(bthData, bthDataLen, this, m_Packet);
+    }
+
+    void InfiniBandLayer::computeCalculateFields()
+	{
+        
+	}
+
+	std::string InfiniBandLayer::toString() const
+	{
+        std::ostringstream opCodeStream;
+        opCodeStream << getOpcode();
+
+        return "InfiniBand Layer, Opcode: " + opCodeStream.str();
+	}
```

Comment:
Using `std::to_string(x)` when converting single values is more efficient then using a `std::stringstream`.